### ISSUE TITLE
Resnet 3D-50

### ISSUE THREAD

**Issue by zwtu** [lambert-x/video-semisup#2]
Hi!
I basically have reproduced the results on the basis of the project.I found that the backbone used in the paper are resnet 18 and I replaced it with resnet 50 or vit-s on the basic framework of Fixmatch, but neither seems to work well.
I don't know if you have done similar training, so that the effect is inherently bad, or if there is a problem with my training itself.

**Comment by lambert-x** [comment]
ViT features are much different from CNNs', so you might need heavy modifications to make the method work.  Using a different backbone like ResNet-50 usually needs further parameters tuning (e.g., learning rate, warmup iterations). By the way, I don't if you have reduced the batchsize to avoid GPU out-of-memory. If yes, this would be harmful to the contrastive learning part as it requires a large batchsize to get enough negative pairs.

**Comment by zwtu** [comment]
I noticed that in the 10% labeled UCF-101 setting, the FixMatch result in the paper is 38.4% when using Resnet18 as the backbone. A concurrent work(CMPL) using Resnet50 resulted in 49.8%. I wonder if a backbone replacement, by further parameters tuning, can bring such a big improvement.
Many thanks!

**Comment by lambert-x** [comment]
> I noticed that in the 10% labeled UCF-101 setting, the FixMatch result in the paper is 38.4% when using Resnet18 as the backbone. A concurrent work(CMPL) using Resnet50 resulted in 49.8%. I wonder if a backbone replacement, by further parameters tuning, can bring such a big improvement. Many thanks!

What result have you got by using ResNet-50 with my codebase? I will try to produce the result you mentioned (ResNet50-FixMatch: 49.8%) in the next serveral days. By the way, there are multiple RandAug (for video clips) implementations publicly available and I am not sure if this will make a big difference.

**Comment by zwtu** [comment]
First of all thank you very much indeed for opening the source code. This article is [CMPL (CVPR 2022)](https://openaccess.thecvf.com/content/CVPR2022/html/Xu_Cross-Model_Pseudo-Labeling_for_Semi-Supervised_Action_Recognition_CVPR_2022_paper.html).It is a great pity that CMPL does not disclose the code, but his result is SOTA, surpassing MvPL (ICCV 2021).I tried to reproduce CMPL on top of your code, but the result were really bad. I was able to almost reproduce your result. 
BTW, in the 10% labeled UCF-101 setting, the result of MvPL using Slowonly with RGB and TG is 74.0%. Meanwhile, the result of your reimplemented MvPL using R3D-18 is 55.5%. Have you used slowonly for reimplementation? I am now troubled by the fact that I am having a hard time reproducing nearly the same results following the experimental details of these non-open source papers.
Thank you for always replying to me！

**Comment by lambert-x** [comment]
> First of all thank you very much indeed for opening the source code. This article is [CMPL (CVPR 2022)](https://openaccess.thecvf.com/content/CVPR2022/html/Xu_Cross-Model_Pseudo-Labeling_for_Semi-Supervised_Action_Recognition_CVPR_2022_paper.html).It is a great pity that CMPL does not disclose the code, but his result is SOTA, surpassing MvPL (ICCV 2021).I tried to reproduce CMPL on top of your code, but the result were really bad. I was able to almost reproduce your result. BTW, in the 10% labeled UCF-101 setting, the result of MvPL using Slowonly with RGB and TG is 74.0%. Meanwhile, the result of your reimplemented MvPL using R3D-18 is 55.5%. Have you used slowonly for reimplementation? I am now troubled by the fact that I am having a hard time reproducing nearly the same results following the experimental details of these non-open source papers. Thank you for always replying to me！

Two config files for the MvPL reimplementation are uploaded in https://github.com/lambert-x/video-semisup/tree/main/configs/ucf101_MvPL.  The results I got are 49.5 (48.5 paper) and 71 (74.0 paper). The reason for the performance gap might be different data split strategy and batch size. You are welcome to polish the codes and update the reimplementation. Good luck!
